x-pack/plugins/apm/server/lib/helpers/get_random_sampler/index.ts

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { KibanaRequest } from '@kbn/core/server';
+import seedrandom from 'seedrandom';
+import { APMRouteHandlerResources } from '../../..';
+
+export async function getRandomSampler({
+  security,
+  request,
+  probability,
+}: {
+  security: APMRouteHandlerResources['plugins']['security'];
+  request: KibanaRequest;
+  probability: number;
+}) {
+  const defaultSeed = 1;
```

Comment:
Any other suggestion?

Reply:
Btw. What's the downsides of hardcoding the seed (perhaps making it configurable via advanced settings) and not using the username at all?

It's probably a bad idea but I'm interested to know why.

Reply:
@sqren biased results presumably?